internal/storage/writer/talaria/talaria.go

```diff
@@ -0,0 +1,109 @@
+package talaria
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	talaria "github.com/kelindar/talaria/client/golang"
+	"github.com/kelindar/talaria/internal/encoding/key"
+	"github.com/kelindar/talaria/internal/monitor/errors"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+// GetClient will create a Talaria client
+func GetClient(endpoint string, dialTimeout time.Duration, circuitTimeout time.Duration, maxConcurrent int, errorThresholdPercent int) (*talaria.Client, error) {
+
+	var client *talaria.Client
+	var err error
+
+	client, err = talaria.Dial(endpoint, talaria.WithNetwork(dialTimeout), talaria.WithCircuit(circuitTimeout, maxConcurrent, errorThresholdPercent))
+
+	if err != nil {
+		return nil, err
+	}
+	return client, nil
+}
+
+// Writer to write to TalariaDB
+type Writer struct {
+	lock                  sync.Mutex
+	endpoint              string
+	dialTimeout           time.Duration
+	circuitTimeout        time.Duration
+	maxConcurrent         int
+	errorPercentThreshold int
+	client                *talaria.Client
+}
+
+// New initializes a new Talaria writer.
+func New(endpoint string, dialTimeout time.Duration, circuitTimeout time.Duration, maxConcurrent int, errorPercentThreshold int) (*Writer, error) {
+
+	dialTimeout = dialTimeout * time.Second
+	circuitTimeout = circuitTimeout * time.Second
```

Comment:
Could we have sensible defaults here? I think it should be rare to require configuration for any of these parameters.